src/directives/pagination.ts

```diff
@@ -64,16 +63,15 @@ export const paginationDirective = <T>({table}: PaginationDirectiveConfiguration
             return pageCount > currentPage;
         },
         state() {
-            return Object.assign(table.getTableState().slice, {filteredCount: itemListLength, pageCount});
+            return Object.assign(table.getTableState().slice, {pageCount});
         }
     };
     const directive = Object.assign(api, proxy);
 
     directive.onSummaryChange(({page: p, size: s, filteredCount}: Summary) => {
         currentPage = p;
         currentSize = s;
-        itemListLength = filteredCount;
-        pageCount = currentSize ? Math.ceil(itemListLength / currentSize) : 1;
+        pageCount = currentSize && filteredCount ? Math.ceil(filteredCount / currentSize) : 1;
```

Comment:
Maybe pageCount should be part of the summary and/or be a table instance property. It's useful in a summary directive.